app/cli/internal/action/attestation_status.go

```diff
@@ -163,16 +221,35 @@ func pbAnnotationsToAction(in []*pbc.Annotation) []*Annotation {
 	return res
 }
 
-func setMaterialValue(w *v1.Attestation_Material, o *Material) error {
+// stateAnnotationToAction converts the map of annotations to a slice of []*Annotation
+func stateAnnotationToAction(in map[string]string) []*Annotation {
+	res := make([]*Annotation, 0, len(in))
+
+	for k, v := range in {
+		res = append(res, &Annotation{
+			Name:  k,
+			Value: v,
+		})
+	}
+
+	return res
+}
+
+func setMaterialValue(w *v1.Attestation_Material, o *AttestationStatusResultMaterial) error {
+	setCommonFields := func(value, hash, tag string) {
```

Comment:
Wouldn't be more Go idiomatic to just mutate the struct like this at the top of the function?

```
o.Set = true
o.Tag = v.GetContainerImage().GetTag()
```

The current implementation is strange because it seems it's meant to give you control on what to pass depending on the value, but then you do not use that control (you are always passing the `getContainerImage().GetTag()`

Reply:
Sure thing, I'll add to the bottom to avoid nil pointers.